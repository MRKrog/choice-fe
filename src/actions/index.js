export const setLoading = loading => ({
  type: "SET_LOADING",
  loading
});

export const setOrders = orders => ({
  type: "SET_ORDERS",
  orders
});

export const setNotes = notes => ({
  type: "SET_NOTES",
  notes
});
