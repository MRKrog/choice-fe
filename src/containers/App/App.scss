.App {
  width: 100%;
  height: 100%;
  display: flex;
  flex-direction: column;

  .Content-Container {
    display: flex;
    height: 100%;
    position: relative;
    z-index: 9;
  }
}
