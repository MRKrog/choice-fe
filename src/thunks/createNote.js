import * as actions from '../actions/index';

export const createNote = (currentOrder, notes) => {
  return (dispatch) => {
    let sortedNotes = notes.sort((a,b) => a.id - b.id)
    let lastNumber = (sortedNotes[sortedNotes.length - 1].id) + 1
    let newNote = {
      title: "Enter Title",
      copy: "Enter Copy",
      status: 0,
      order_id: currentOrder,
      id: lastNumber,
      notSaved: true
    }
    dispatch(actions.setCurrNote(lastNumber));
    dispatch(actions.createNewNote(newNote));
  }
}
