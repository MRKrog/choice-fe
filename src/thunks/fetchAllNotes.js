import * as actions from '../actions/index';
import { fetchData } from '../utility/fetchData';

export const fetchAllNotes = () => {
  return async (dispatch) => {
    try {
      dispatch(actions.setLoading(true));
      const allNotes = await fetchData(process.env.REACT_APP_BACKEND_URL + '/api/v1/notes');
      let sortedNotes = allNotes.sort((a,b) => a.id - b.id)
      dispatch(actions.setNotes(sortedNotes));
      dispatch(actions.setLoading(false));
    } catch (error) {
      dispatch(actions.setError(error.message));
    }
  }
}
