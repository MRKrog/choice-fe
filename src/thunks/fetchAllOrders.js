import * as actions from '../actions/index';
import { fetchData } from '../utility/fetchData';

export const fetchAllOrders = () => {
  return async (dispatch) => {
    try {
      dispatch(actions.setLoading(true));
      const allOrders = await fetchData(process.env.REACT_APP_BACKEND_URL + '/api/v1/orders');
      dispatch(actions.setOrders(allOrders));
      // dispatch(actions.setLoading(false));
    } catch (error) {
      dispatch(actions.setError(error.message));
    }
  }
}
