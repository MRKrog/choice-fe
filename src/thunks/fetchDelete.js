import * as actions from '../actions/index';
import { fetchData } from '../utility/fetchData';
import { fetchOptions } from '../utility/fetchOptions';

export const fetchDelete = (url) => {
  return async (dispatch) => {
    try {
      dispatch(actions.setLoading(true));
      const options = await fetchOptions('DELETE', {});
      const response = await fetchData(url, options);
      if(!response.ok){
        throw Error(response.statusText)
      }
      dispatch(actions.setLoading(false));
    } catch (error) {
      dispatch(actions.setError(error.message));
    }
  }
}
