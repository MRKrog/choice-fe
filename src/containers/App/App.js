import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import * as actions from '../../actions';

import Header from '../Header/Header';
import SideBar from '../SideBar/SideBar';
import NoteContainer from '../NoteContainer/NoteContainer';
import NoteDisplay from '../NoteDisplay/NoteDisplay';
import Loading from '../../components/Loading/Loading';

import { fetchAllOrders } from '../../thunks/fetchAllOrders';
import { fetchAllNotes } from '../../thunks/fetchAllNotes';

export class App extends Component {

  async componentDidMount() {
    const { fetchAllOrders, fetchAllNotes } = this.props;
    await fetchAllOrders();
    await fetchAllNotes();
  }

  render() {
    const { currentOrder, currentNote } = this.props;
    const currNoteTabs = this.props.notes.filter(note => note.order_id === currentOrder);
    const clickedNote = this.props.notes.find(note => note.id === currentNote);

    return (
      <div className='App'>
      <Header />
      {
        this.props.loading
        ? <Loading />
        : (
          <section className="Content-Container">
            {
              this.props.orders.length &&
              <SideBar />
            }
            {
              this.props.notes.length &&
              <NoteContainer currNoteTabs={currNoteTabs} key={`${currentOrder}Order`}/>
            }
            {
              typeof this.props.currentNote === 'number' &&
              <NoteDisplay {...clickedNote} key={clickedNote.id}/>
            }
          </section>
        )
      }
      </div>
    )
  }
}



export const mapStateToProps = state => ({
  loading: state.loading,
  orders: state.orders,
  notes: state.notes,
  currentOrder: state.currentOrder,
  currentNote: state.currentNote,
});

export const mapDispatchToProps = dispatch => ({
  setLoading: data => dispatch(actions.setLoading(data)),
  setOrders: data => dispatch(actions.setOrders(data)),
  setNotes: data => dispatch(actions.setNotes(data)),
  setCurrOrder: data => dispatch(actions.setCurrOrder(data)),
  fetchAllOrders: data => dispatch(fetchAllOrders(data)),
  fetchAllNotes: data => dispatch(fetchAllNotes(data)),
});

App.propTypes = {
  loading: PropTypes.bool,
}

export default connect(mapStateToProps, mapDispatchToProps)(App);
