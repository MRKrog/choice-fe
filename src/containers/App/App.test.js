import React from 'react';
import { shallow, mount } from 'enzyme'
import { App, mapStateToProps, mapDispatchToProps } from './App';

import { fetchAllOrders } from '../../thunks/fetchAllOrders';
import { fetchAllNotes } from '../../thunks/fetchAllNotes';

jest.mock('../../thunks/fetchAllOrders');
jest.mock('../../thunks/fetchAllNotes');

import * as actions from '../../actions/index';


const mockNotes = [
  {
    id: 1,
    title: 'Note Title 0101',
    copy: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit',
    status: 0,
    order_id: 9
  },
  {
    id: 2,
    title: 'Note Title 0102',
    copy: 'Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium',
    status: 0,
    order_id: 8
  },
  {
    id: 3,
    title: 'Note Title 0103',
    copy: 'Ut enim ad minima veniam, quis nostrum exercitationem ullam corporis suscipit laboriosam, nisi ut aliquid ex ea commodi consequatur?',
    status: 1,
    order_id: 9
  }
]

const mockOrders = [
  {
    id: 8,
    number: 2222,
  },
  {
    id: 9,
    number: 2222,
  }
]

describe('App', () => {

  describe('App Component', () => {
    let wrapper;
    let mockfetchAllOrders = jest.fn()
    let mockfetchAllNotes = jest.fn()

    beforeEach(() => {
      wrapper = shallow(<App fetchAllOrders={mockfetchAllOrders}
                             fetchAllNotes={mockfetchAllNotes}
                             notes={mockNotes}
                             orders={mockOrders}
                             />
                        )
    })

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it("should invoke the thunk fetchAllOrders on componentDidMount", async () => {
      await wrapper.instance().componentDidMount()
      expect(mockfetchAllOrders).toBeCalled()
    })

    it("should invoke the thunk fetchAllNotes on componentDidMount", async () => {
      await wrapper.instance().componentDidMount()
      expect(mockfetchAllNotes).toBeCalled()
    })

  });

  describe('mapStateToProps', () => {

  });

  describe('mapDispatchToProps', () => {

  });

});
