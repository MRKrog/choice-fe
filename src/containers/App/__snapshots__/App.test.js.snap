// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App Component should match the snapshot 1`] = `
<div
  className="App"
>
  <Connect(Header) />
  <section
    className="Content-Container"
  >
    <Connect(SideBar) />
    <Connect(NoteContainer)
      currNoteTabs={Array []}
      key="undefinedOrder"
    />
  </section>
</div>
`;
